src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySetSource.java

```diff
@@ -52,36 +53,46 @@ default boolean isGeneralNoThroughTraffic(OSMWithTags way) {
 
   default boolean isVehicleThroughTrafficExplicitlyDisallowed(OSMWithTags way) {
     String vehicle = way.getTag("vehicle");
-    return isGeneralNoThroughTraffic(way) || doesTagValueDisallowThroughTraffic(vehicle);
+    if (vehicle != null) {
+      return doesTagValueDisallowThroughTraffic(vehicle);
+    } else {
+      return isGeneralNoThroughTraffic(way);
+    }
   }
 
   /**
    * Returns true if through traffic for motor vehicles is not allowed.
    */
   default boolean isMotorVehicleThroughTrafficExplicitlyDisallowed(OSMWithTags way) {
     String motorVehicle = way.getTag("motor_vehicle");
-    return (
-      isVehicleThroughTrafficExplicitlyDisallowed(way) ||
-      doesTagValueDisallowThroughTraffic(motorVehicle)
-    );
+    if (motorVehicle != null) {
+      return doesTagValueDisallowThroughTraffic(motorVehicle);
+    } else {
+      return isVehicleThroughTrafficExplicitlyDisallowed(way);
+    }
   }
 
   /**
    * Returns true if through traffic for bicycle is not allowed.
    */
   default boolean isBicycleNoThroughTrafficExplicitlyDisallowed(OSMWithTags way) {
     String bicycle = way.getTag("bicycle");
-    return (
-      isVehicleThroughTrafficExplicitlyDisallowed(way) ||
-      doesTagValueDisallowThroughTraffic(bicycle)
-    );
+    if (bicycle != null) {
+      return doesTagValueDisallowThroughTraffic(bicycle) || "dismount".equals(bicycle);
```

Comment:
Would not make sense to add `dismount` to `OSMWithTags#isBicycleExplicitlyDenied`? I think that's also more correct.

Reply:
Yes, it would :100: - I've moved it.